cEP-0010.md

```diff
@@ -0,0 +1,334 @@
+with the Root aspect at the root of the tree. Some classes, metaclasses
+and some few methods were define so far to make that possible including:
+
+* [the aspectbase class](https://github.com/coala/coala/blob/master/coalib/bearlib/aspects/base.py)
+* [the Documentation class for documenting aspects](https://github.com/coala/coala/blob/master/coalib/bearlib/aspects/docs.py)
+* [the aspectclass metaclass](https://github.com/coala/coala/blob/master/coalib/bearlib/aspects/meta.py)
+* [the TasteMeta class](https://github.com/coala/coala/blob/master/coalib/bearlib/aspects/taste.py)
+* [the Taste class](https://github.com/coala/coala/blob/master/coalib/bearlib/aspects/taste.py)
+
+`aspects` will simply be defined this way:
+```python
+from coalib.bearlib.aspects import Root, Taste
+
+
+@Root.subaspect
+class SomeAspect:
+#This will serve as documentation for the aspect
+    class docs:
+        example = 'an example of this aspect of code'
+        example_language = 'the programming language of the example'
+        importance_reason = 'the reason why this is important'
+        fix_suggestions = ''
+#Here we define the different tastes for this aspect
+#aspects are not compelled to have tastes
+    some_taste = Taste[Taste_type](
+        'Documentation about this taste',
+        list('Possible value for this taste',),
+        default='Default value for this taste')
+```
+The documentation of each of the implemented aspects will be added to
+the [coala/aspect-docs](https://github.com/coala/aspect-docs) repository.
+
+Concerning bears' results annotation, so far, the `.aspect` attr was added to
+the [Result class](https://github.com/coala/coala/blob/master/coalib/results/Result.py)
+(where a Result object is what is yielded by a bear when through executing).
+So annotating a bear's result with aspects simply consists of setting this
+variable to the appropriate aspect of code.
+Here is how bears' will be defined in as presented in
+[cEP-0005](https://github.com/coala/cEPs/blob/master/cEP-0005.md).
+
+```python
+from coalib.bears.LocalBear import LocalBear
+from coalib.results.Result import Result
+from coalib.bearlib.aspects import Root
+
+
+class RedundancyBear(LocalBear, aspects={
+        # this represents all the aspects this bear can detect.
+        'detect': [Root.Redundancy.Clone],
+        # this represents all the aspects this bear can fix.
+        'fix': [Root.Redundancy.UnusedImport,
+        languages=['C', 'Python']):
+
+    def run(self, filename, file, aspects):
+        # No documentation required anymore for the bears.
+
+        if unused_imports(file):
+            # A bear may yield results for any aspect even if it's not selected.
+            # coala will filter out only selected aspects.
+            yield Result.from_values(aspect=Root.Redundancy.UnusedImport, ...)
+
+        # Bears can save performance by only performing needed checks
+        aspect = aspects.get(Root.Redundancy.Clone, False)
```

Comment:
We haven't decided on a data type for the `aspects` argument yet... You are just assuming a dict here that maps aspectclasses to aspect instances... Don't know if that's the best idea! But let's just leave it for now and refine later :)